auctions/english-auction/src/utils.sw

```diff
@@ -0,0 +1,125 @@
+library utils;
+
+dep abi;
+dep data_structures;
+dep errors;
+
+use abi::NFT;
+use data_structures::{Asset, NFTAsset, TokenAsset};
+use errors::{AccessError, InputError};
+use std::{
+    assert::require,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::{contract_id, msg_asset_id}, msg_amount},
+    identity::Identity,
+    option::Option,
+    result::Result,
+    token::transfer,
+};
+
+/// This function returns true if the `to` `Identity` is approved to transfer the token.
```

Comment:
I'm not going to say to remove the docs, yet, but idk if we should be documenting the functions, like this, that are not directly on the ABI. we probably should though... 

can you order the params and the docs in the same order?

Reply:
There can never be enough documentation imo. 

But following what we see in Solidity, any libraries imported have their own documentation as well.